code/datums/status_effects/debuffs/staggered.dm

```diff
@@ -0,0 +1,42 @@
+/datum/status_effect/staggered
+	id = "staggered"
+	tick_interval = 0.5 SECONDS
+	alert_type = null
+	remove_on_fullheal = TRUE
+
+/datum/status_effect/staggered/on_creation(mob/living/new_owner, duration = 10 SECONDS)
+	src.duration = duration
+	return ..()
+
+/datum/status_effect/staggered/on_apply()
+
+	//a very mild animation, but you can't stagger the dead.
+	if(owner.stat == DEAD)
+		owner.do_stagger_animation(duration / 10)
+		return FALSE
+
+	RegisterSignal(owner, COMSIG_LIVING_DEATH, PROC_REF(clear_staggered))
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/staggered)
+	return TRUE
+
+/datum/status_effect/staggered/on_remove()
+	UnregisterSignal(owner, COMSIG_LIVING_DEATH)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/staggered)
+	// Resetting both X and Y positioning when we remove, even though we only animates on the X axis
+	owner.pixel_y = 0
+	owner.pixel_x = 0
+
+/// Signal proc that self deletes our staggered effect
+/datum/status_effect/staggered/proc/clear_staggered(datum/source)
+	SIGNAL_HANDLER
+
+	qdel(src)
+
+/datum/status_effect/staggered/tick(seconds_between_ticks)
+	owner.do_stagger_animation()
+
+/// Helper proc that causes the mob to do a stagger animation.
+/// Doesn't change significantly, just meant to represent swaying back and forth
+/mob/living/proc/do_stagger_animation()
+	animate(src, pixel_x = 4, pixel_y = 0 , time = 0.2 SECONDS, loop = 6, flags = ANIMATION_RELATIVE|ANIMATION_PARALLEL)
+	animate(pixel_x = -4 , pixel_y = 0 , time = 0.2 SECONDS, flags = ANIMATION_RELATIVE)
```

Comment:
same here. but why are you even modifying pixel_y?

Reply:
you know what, good question